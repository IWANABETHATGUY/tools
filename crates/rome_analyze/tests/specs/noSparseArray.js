// valid
const a = [ 1, 2, ];

// invalid
const b = [/**test*/,];
const c = [,,];
const d = [,,2];

const e = [1,,2]