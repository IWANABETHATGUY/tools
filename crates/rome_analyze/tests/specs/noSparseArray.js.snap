---
source: crates/rome_analyze/tests/spec_tests.rs
assertion_line: 96
expression: noSparseArray.js
---
# Input
```js
// valid
const a = [ 1, 2, ];

// invalid
const b = [/**test*/,];
const c = [,,];
const d = [,2];

const e = [1,,]

```

# Diagnostics
```
warning[noSparseArray]: This array contains an empty slot.
  ┌─ noSparseArray.js:5:11
  │
5 │ const b = [/**test*/,];
  │           ------------ This array contains an empty slot.

Replace with undefined
    | @@ -2,7 +2,7 @@
1 1 |   const a = [ 1, 2, ];
2 2 |   
3 3 |   // invalid
4   | - const b = [/**test*/,];
  4 | + const b = [/**test*/undefined,];
5 5 |   const c = [,,];
6 6 |   const d = [,2];
7 7 |   


```

```
warning[noSparseArray]: This array contains an empty slot.
  ┌─ noSparseArray.js:6:11
  │
6 │ const c = [,,];
  │           ---- This array contains an empty slot.

Replace with undefined
    | @@ -3,7 +3,7 @@
2 2 |   
3 3 |   // invalid
4 4 |   const b = [/**test*/,];
5   | - const c = [,,];
  5 | + const c = [undefined,undefined,];
6 6 |   const d = [,2];
7 7 |   
8 8 |   const e = [1,,]


```

```
warning[noSparseArray]: This array contains an empty slot.
  ┌─ noSparseArray.js:7:11
  │
7 │ const d = [,2];
  │           ---- This array contains an empty slot.

Replace with undefined
    | @@ -4,6 +4,6 @@
3 3 |   // invalid
4 4 |   const b = [/**test*/,];
5 5 |   const c = [,,];
6   | - const d = [,2];
  6 | + const d = [undefined,2];
7 7 |   
8 8 |   const e = [1,,]


```

```
warning[noSparseArray]: This array contains an empty slot.
  ┌─ noSparseArray.js:9:11
  │
9 │ const e = [1,,]
  │           ----- This array contains an empty slot.

Replace with undefined
    | @@ -6,4 +6,4 @@
5 5 |   const c = [,,];
6 6 |   const d = [,2];
7 7 |   
8   | - const e = [1,,]
  8 | + const e = [1,undefined,]


```


